test/notebook-tests.py

```diff
@@ -201,93 +200,123 @@ def test_t5():
 
+    # Clean up residual files
+    shutil.rmtree(exp_path)
+
+
+# -------------Tutorial 6-------------------------------------------------------
+"""
+...
+"""
 
-# #-------------Tutorial 6-------------------------------------------------------
-# """
-# ...
-# """
-# def test_t6():
-# # Evaluation
-# model = mlp.choose_model()
-# download_dir = kso_widgets.choose_folder(".", "where to download the model")
-# artifact_dir = mlp.get_model(model.value, download_dir.selected)
-# source = kso_widgets.choose_footage(
-#     project=pp.project,
-#     server_connection=pp.server_connection,
-#     db_connection=pp.db_connection,
-#     start_path=pp.project.movie_folder
-#     if pp.project.movie_folder not in [None, "None"]
-#     else ".",
-#     folder_type="custom footage",
-# )
-# source_value = t_utils.process_source(source)
-# save_dir = kso_widgets.choose_folder(".", "runs output")
-# conf_thres = t_utils.choose_conf()
-# mlp.detect_yolov5(
-#     source=source_value,
-#     save_dir=save_dir.selected,
-#     conf_thres=conf_thres.value,
-#     artifact_dir=artifact_dir,
-# )
-# eval_dir = kso_widgets.choose_folder(
-#     save_dir.selected
-#     if "save_dir" in vars() and save_dir.selected is not None
-#     else ".",
-#     "runs output",
-# )
-# mlp.save_detections_wandb(conf_thres.value, model.value, eval_dir.selected)
-# viewer = y_utils.choose_files(eval_dir.selected)
-# train_dataset, val_dataset = mlp.get_dataset(model.value)
-# y_utils.get_data_viewer(os.path.join(train_dataset, "data/images"))
-# y_utils.get_data_viewer(os.path.join(val_dataset, "data/images"))
-
-# # Tracking individuals
-# mlp.track_individuals(
-#     source=source_value,
-#     artifact_dir=artifact_dir,
-#     eval_dir=eval_dir.selected,
-#     conf_thres=conf_thres.value,
-#     img_size=(540, 540),
-# )
+
+def test_t6():
+    import kso_utils.tutorials_utils as t_utils
+    import kso_utils.server_utils as s_utils
+
+    # Generate current timestamp
+    import shutil
+    from datetime import datetime
+
+    dt = datetime.now()
+    # Create a unique experiment name
+    exp_name = f"custom_{dt}".replace(" ", "_").replace(".", "_").replace(":", "-")
+    mlp.output_path = "../test/test_output"
+    project_path = os.path.join(mlp.output_path, mlp.project_name)
+    exp_path = os.path.join(project_path, exp_name)
+    # Evaluation
+    s_utils.get_ml_data(project, test=True)
+    model = mlp.choose_model().options[-1][1]
+    download_dir = mlp.output_path
+    artifact_dir = mlp.get_model(model, download_dir)
+    source = os.path.join("../test/test_output", mlp.project.ml_folder, "images")
+    save_dir = project_path
+    conf_thres = t_utils.choose_conf()
+    mlp.detect_yolov5(
+        exp_name=exp_name,
+        source=source,
+        save_dir=save_dir,
+        conf_thres=conf_thres.value,
+        artifact_dir=artifact_dir,
+    )
+    eval_dir = exp_path
+    mlp.save_detections_wandb(conf_thres.value, model, eval_dir)
+
```

Comment:
Maybe put a comment that the 'Investigating training and validation dataset' is not tested, and that we now go to track unique individuals.